app/services/cdl_checkout.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+# Procedurual service for checking out CDL content from symphony
+class CdlCheckout
+  attr_reader :barcode, :druid, :user
+
+  def initialize(barcode, druid, user)
+    @barcode = barcode
+    @druid = druid
+    @user = user
+  end
+
+  ##
+  # @return token [String]
+  def process_checkout
+    place_hold
+    checkout = place_checkout
+
+    due_date = checkout&.dig('circRecord', 'fields', 'dueDate')
+    error_messages = Array.wrap(checkout&.dig('messageList')).map { |message| message.dig('message') }
+
+    raise(Exceptions::CdlCheckoutError, error_messages.join(' ')) if error_messages.present?
+
+    # TODOS:
+    # schedule a job to check the item back in?
+    # schedule a job to remove the users hold on the item
+    create_token(due_date)
+  end
+
+  def create_token(due_date)
+    payload = {
+      barcode: selected_barcode,
+      aud: druid,
+      sub: user.webauth,
+      exp: DateTime.parse(due_date).to_i
+    }
+    JWT.encode(payload, Settings.cdl.jwt.secret, Settings.cdl.jwt.algorithm)
+  end
+
+  def place_hold
+    symphony_client.place_hold(
+      comment: "CDL checkout for #{druid}",
+      fill_by_date: DateTime.now + 1.year,
+      patron_barcode: user.library_id,
+      item: {
+        call: { key: callkey, resource: '/catalog/call' },
+        holdType: 'TITLE'
+      }
+      # key: 'CDL' # TODO: What should this be?
```

Comment:
Just so we remember:

```suggestion
      key: 'SUL'
```